bundles/core/org.eclipse.smarthome.core.ephemeris/src/main/java/org/eclipse/smarthome/core/ephemeris/internal/EphemerisManagerImpl.java

```diff
@@ -0,0 +1,237 @@
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.jollyday.Holiday;
+import de.jollyday.HolidayManager;
+import de.jollyday.ManagerParameter;
+import de.jollyday.ManagerParameters;
+
+/**
+ * This service provides functionality around ephemeris services and is the central service to be used directly by
+ * others.
+ *
+ * @author Gaël L'hopital - Initial contribution and API
+ *
+ */
+@Component(immediate = true, configurationPid = "org.eclipse.smarthome.ephemeris", property = { //
+        Constants.SERVICE_PID + "=org.eclipse.smarthome.ephemeris", //
+        ConfigurableService.SERVICE_PROPERTY_CATEGORY + "=system", //
+        ConfigurableService.SERVICE_PROPERTY_LABEL + "=Ephemeris", //
+        ConfigurableService.SERVICE_PROPERTY_DESCRIPTION_URI + "=" + EphemerisManagerImpl.CONFIG_URI //
+})
+public class EphemerisManagerImpl implements EphemerisManager, ConfigOptionProvider {
+    private final static Logger logger = LoggerFactory.getLogger(EphemerisManagerImpl.class);
+
+    // constants for the configuration properties
+    protected static final String CONFIG_URI = "system:ephemeris";
+    private static final String CONFIG_WEEKENDAYS = "weekendDays";
+    private static final String CONFIG_COUNTRY = "country";
+    private static final String CONFIG_REGION = "region";
+    private static final String CONFIG_CITY = "city";
+
+    private final Set<DayOfWeek> weekend = new HashSet<>();
+    private final HashMap<Object, HolidayManager> holidayManagers = new HashMap<>();
+    private String country;
+    private final ArrayList<String> countryParameters = new ArrayList<>();
+
+    private LocaleProvider localeProvider;
+
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        modified(config);
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        String weekEndDays = config.containsKey(CONFIG_WEEKENDAYS) ? config.get(CONFIG_WEEKENDAYS).toString() : null;
+        if (weekEndDays != null) {
+            String[] days = weekEndDays.split(",");
+            for (String day : days) {
+                weekend.add(DayOfWeek.valueOf(day.toUpperCase()));
+            }
+        } else {
+            logger.info("No days were configured as week-end days, you'll have to work all week");
+        }
+
+        country = config.containsKey(CONFIG_COUNTRY) ? config.get(CONFIG_COUNTRY).toString() : null;
+        if (country != null) {
+            logger.info("Default country '{}' overridden by configured one '{}'",
+                    localeProvider.getLocale().getCountry(), country);
+        } else {
+            country = localeProvider.getLocale().getCountry();
+            logger.info("Using system default country '{}' ", country);
```

Comment:
Is al this logging important enough to be logged at info instead of debug? See [Logging guidelines](https://www.eclipse.org/smarthome/documentation/development/guidelines.html#e-logging).